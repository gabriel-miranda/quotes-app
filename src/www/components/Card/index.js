import React from 'react';
import PropTypes from 'prop-types';

const CARD_STYLES = {
  margin: '20px 0',
  padding: '20px',
  borderRadius: '3px',
};

const Card = ({id, text, authorName}) => (
  <div className="card-docs" style={CARD_STYLES}>
    <p>Quote <strong>#{id}</strong></p>
    <p>"{text}"</p>
    <p>
      <strong>
        <em>
          - {authorName || 'Anonymous'}
        </em>
      </strong>
    </p>
  </div>
);

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  authorName: PropTypes.string.isRequired,
};

export default Card;
