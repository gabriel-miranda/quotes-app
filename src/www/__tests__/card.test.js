/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../components/Card';

const cardWithAuthor = {
  id: 23,
  text: 'This is a test quote',
  authorName: 'Gabriel',
};

const cardWithoutAuthor = {
  id: 24,
  text: 'This is a test quote without author',
  authorName: '',
};

describe('Card test with enzyme', () => {
  it('should show authorName correctly', () => {
    const card = shallow(<Card {...cardWithAuthor}/>);
    expect(card.find('em').text()).toEqual(cardWithAuthor.authorName);
  });
  it('should show Anonymous if no author is provided', () => {
    const card = shallow(<Card {...cardWithoutAuthor}/>);
    expect(card.find('em').text()).toEqual('Anonymous');
  });
});

describe('Header test with Snapshot Testing', () => {
  it('should show Card correctly', () => {
    const component = renderer.create(<Card  {...cardWithAuthor}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
