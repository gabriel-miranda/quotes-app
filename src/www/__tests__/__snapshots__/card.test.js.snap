// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test with Snapshot Testing should show Card correctly 1`] = `
<div
  className="card-docs"
  style={
    Object {
      "borderRadius": "3px",
      "margin": "20px 0",
      "padding": "20px",
    }
  }
>
  <p>
    Quote 
    <strong>
      #
      23
    </strong>
  </p>
  <p>
    "
    This is a test quote
    "
  </p>
  <p>
    <strong>
      -
      <em>
        Gabriel
      </em>
    </strong>
  </p>
</div>
`;
